### PR TITLE
Chat push nav

### PR DESCRIPTION
**Pull request by gprasant**
1. Fixes some troublesome UI issues we were seeing (On the ChatsVC and the ChatVC )
   (Does not need the Margin hacks now. )
2. Hides Heart and X button on the profile View because it will need too much work to get it working at this point 
3. Push navigation from the CardsVC -> Chats VC
4. Push navigation from the ChatsVC -> ChatVC

### CODE REVIEW

**Comment by gprasant** [comment]
Adding a smoke_test

smoke_test

File: ChatViewController.m [hunk 1]
```diff
@@ -43,13 +44,21 @@ - (void) viewDidLoad {
...
+}
+
 - (void) didReceiveMemoryWarning {
     [super didReceiveMemoryWarning];
     // Dispose of any resources that can be recreated.
 }
 
 - (IBAction) onBackTapped:(id)sender {
-    [self dismissViewControllerAnimated:YES completion:nil];
+    [self.navigationController popViewControllerAnimated:YES];
```
Comment (clusterwin) [inline ChatViewController.m:57]: Why does the dismiss no longer get called?

**Comment by greysongregory** [comment]
merging so i can make some final touches

File: ChatViewController.m (hunk 1 above)
Comment (clusterwin) [inline ChatViewController.m:57]: @gprasant

Comment (gprasant) [inline ChatViewController.m:57]: Because with a push navigation, popViewControllerAnimated is the equivalent of dismissViewController

**Pull request merged**